File: Actividad 2/Ejercicio18.java
import java.util.Scanner;

class Empleado {
    private int codigoEmpleado;
    private String nombres;
    private int horasTrabajadas;
    private double valorHora;
    private double retencion;

    public Empleado(int codigoEmpleado, String nombres, int horasTrabajadas, double valorHora, double retencion) {
        this.codigoEmpleado = codigoEmpleado;
        this.nombres = nombres;
        this.horasTrabajadas = horasTrabajadas;
        this.valorHora = valorHora;
        this.retencion = retencion;
    }

    public int getCodigoEmpleado() {
        return codigoEmpleado;
    }

    public String getNombres() {
        return nombres;
    }

    public double calcularSalarioBruto() {
        return horasTrabajadas * valorHora;
    }

    public double calcularSalarioNeto() {
        double salarioBruto = calcularSalarioBruto();
        return salarioBruto - (salarioBruto * retencion / 100);
    }

    public void mostrarInformacion() {
        System.out.println("Código del empleado: " + codigoEmpleado);
        System.out.println("Nombres: " + nombres);
        System.out.println("Salario Bruto: $" + calcularSalarioBruto());
        System.out.println("Salario Neto: $" + calcularSalarioNeto());
    }
}

public class Ejercicio18 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.println("Ingrese el código del empleado:");
        int codigo = scanner.nextInt();
        scanner.nextLine(); // Consume newline
        System.out.println("Ingrese los nombres del empleado:");
        String nombres = scanner.nextLine();
        System.out.println("Ingrese el número de horas trabajadas al mes:");
        int horas = scanner.nextInt();
        System.out.println("Ingrese el valor por hora trabajada:");
        double valorHora = scanner.nextDouble();
        System.out.println("Ingrese el porcentaje de retención en la fuente:");
        double retencion = scanner.nextDouble();

        Empleado empleado = new Empleado(codigo, nombres, horas, valorHora, retencion);
        empleado.mostrarInformacion();

        scanner.close();
    }
}

File: Actividad 3/parte_1/Cap_3/ejercicio_18.py
import tkinter as tk

class Empleado:
    def __init__(self, codigo, nombres, horas_trabajadas, valor_hora, porcentaje_retencion):
        self.codigo = codigo
        self.nombres = nombres
        self.horas_trabajadas = horas_trabajadas
        self.valor_hora = valor_hora
        self.porcentaje_retencion = porcentaje_retencion

    def calcular_salario_bruto(self):
        return self.horas_trabajadas * self.valor_hora

    def calcular_retencion(self):
        return self.calcular_salario_bruto() * self.porcentaje_retencion / 100

    def calcular_salario_neto(self):
        return self.calcular_salario_bruto() - self.calcular_retencion()

def calcular_salario_neto():
    codigo = entrada_codigo.get()
    nombres = entrada_nombres.get()
    horas_trabajadas = float(entrada_horas_trabajadas.get())
    valor_hora = float(entrada_valor_hora.get())
    porcentaje_retencion = float(entrada_porcentaje_retencion.get())

    empleado = Empleado(codigo, nombres, horas_trabajadas, valor_hora, porcentaje_retencion)
    salario_neto = empleado.calcular_salario_neto()

    etiqueta_resultado.config(text=f"Salario Neto: ${salario_neto:.2f}")

# Creación de la interfaz gráfica
ventana = tk.Tk()
ventana.title("Calculadora de Salario Neto")

# Campos de entrada
tk.Label(ventana, text="Código del Empleado:").grid(row=0, column=0)
entrada_codigo = tk.Entry(ventana)
entrada_codigo.grid(row=0, column=1)

tk.Label(ventana, text="Nombres:").grid(row=1, column=0)
entrada_nombres = tk.Entry(ventana)
entrada_nombres.grid(row=1, column=1)

tk.Label(ventana, text="Horas Trabajadas:").grid(row=2, column=0)
entrada_horas_trabajadas = tk.Entry(ventana)
entrada_horas_trabajadas.grid(row=2, column=1)

tk.Label(ventana, text="Valor Hora:").grid(row=3, column=0)
entrada_valor_hora = tk.Entry(ventana)
entrada_valor_hora.grid(row=3, column=1)

tk.Label(ventana, text="Porcentaje Retención:").grid(row=4, column=0)
entrada_porcentaje_retencion = tk.Entry(ventana)
entrada_porcentaje_retencion.grid(row=4, column=1)

# Botón para calcular el salario neto
boton_calcular = tk.Button(ventana, text="Calcular Salario Neto", command=calcular_salario_neto)
boton_calcular.grid(row=5, column=0, columnspan=2)

# Etiqueta para mostrar el resultado
etiqueta_resultado = tk.Label(ventana, text="")
etiqueta_resultado.grid(row=6, column=0, columnspan=2)

# Iniciar la ventana principal
ventana.mainloop()
